cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -253,28 +253,31 @@ func deploymentNameForEnv(envName string, clock clock.Clock) string {
 
 // Provisioning the infrastructure within the specified template
 func (p *BicepProvider) Deploy(ctx context.Context, pd *DeploymentPlan) (*DeployResult, error) {
-	done := make(chan bool)
-
-	// Ensure the done marker channel is sent in all conditions
-	defer func() {
-		done <- true
-	}()
-
 	bicepDeploymentData := pd.Details.(BicepDeploymentDetails)
 
-	// Report incremental progress
+	cancelProgress := make(chan bool)
+	defer func() { cancelProgress <- true }()
 	go func() {
+		// Disable reporting progress if needed
+		if v, has := os.LookupEnv("AZD_DEBUG_PROVISION_PROGRESS_DISABLE"); has {
```

Comment:
FWIW - You can use the fact that os.Getenv() returns an empty string (which will not parse for `strconv.ParseBool`, to save yourself a line of indenting by just using os.GetEnv().  Since you check for a nil err coming back from strconv.ParseBool anyway.